🐛 Bug: Lint complaint on missing author on --preset everything

### Bug Report Checklist

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have pulled the latest `main` branch of the repository.
- [x] I have [searched for related issues](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

Creating a new app with `npx create-typescript-app --preset everything` shouldn't produce any errors.

### Actual

```shell
npx create-typescript-app --offline --preset everything --repository cta-temp-oct-27 --description "temp" --title "Temp Oct 27"
┌  ✨ create-typescript-app@2.46.3 ✨
│
●  Learn more on:
│    https://github.com/JoshuaKGoldberg/create-typescript-app
│
◇  Running with mode --setup
│
│  --offline enabled. You'll need to git push any changes manually.
│
◇  Inferred default options from system
│
◇  Ran the Create TypeScript App template
│
◇  Prepared local repository
│
│  You've got a new repository ready to use in:
│    ./cta-temp-oct-27
│
■  The pnpm install --offline --no-frozen-lockfile script failed. You should re-run it and fix its complaints.
│  ExecaError: Command failed with exit code 1: pnpm install --offline --no-frozen-lockfile
│  
│  Progress: resolved 1, reused 0, downloaded 0, added 0
│  \u2009WARN\u2009 deprecated eslint-plugin-markdown@5.1.0: Please use @eslint/markdown instead
│  \u2009ERR_PNPM_NO_MATCHING_VERSION\u2009 No matching version found for @typescript-eslint/typescript-estree@8.46.1
│  
│  This error happened while installing the dependencies of @vitest/eslint-plugin@1.2.7
│   at @typescript-eslint/utils@8.46.1
│  
│  The latest release of @typescript-eslint/typescript-estree is "8.46.0".
│  
│  Other releases are:
│    * rc-v3: 3.0.0-alpha.27
│    * rc-v4: 4.0.0-alpha.16
│    * rc-v5: 5.0.0-alpha.42
│    * rc-v: 6.0.0-alpha.58
│    * rc-v6: 7.0.0-alpha.0
│    * rc-v8: 8.0.0-alpha.62
│    * canary: 8.46.1-alpha.3
│  
│  If you need the full list of all 4406 published versions run "$ pnpm view @typescript-eslint/typescript-estree versions".
│      at getFinalError (file:///Users/josh/repos/create-typescript-app/node_modules/.pnpm/execa@9.6.0/node_modules/execa/lib/return/final-error.js:6:9)
│      at makeError (file:///Users/josh/repos/create-typescript-app/node_modules/.pnpm/execa@9.6.0/node_modules/execa/lib/return/result.js:108:16)
│      at getAsyncResult (file:///Users/josh/repos/create-typescript-app/node_modules/.pnpm/execa@9.6.0/node_modules/execa/lib/methods/main-async.js:168:4)
│      at handlePromise (file:///Users/josh/repos/create-typescript-app/node_modules/.pnpm/execa@9.6.0/node_modules/execa/lib/methods/main-async.js:151:17)
│      at async runCommand (file:///Users/josh/repos/create-typescript-app/node_modules/.pnpm/bingo@0.5.16/node_modules/bingo/lib/runners/applyScriptsToSystem.js:10:24)
│      at async file:///Users/josh/repos/create-typescript-app/node_modules/.pnpm/bingo@0.5.16/node_modules/bingo/lib/runners/applyScriptsToSystem.js:22:17
│      at async Promise.all (index 0)
│      at async applyScriptsToSystem (file:///Users/josh/repos/create-typescript-app/node_modules/.pnpm/bingo@0.5.16/node_modules/bingo/lib/runners/applyScriptsToSystem.js:20:9)
│      at async Promise.all (index 0)
│      at async runCreation (file:///Users/josh/repos/create-typescript-app/node_modules/.pnpm/bingo@0.5.16/node_modules/bingo/lib/runners/runCreation.js:8:5)
│
■  The pnpm lint --fix script failed. You should re-run it and fix its complaints.
│  ExecaError: Command failed with exit code 1: pnpm lint --fix
│  
│  
│  > cta-temp-oct-27@0.0.0 lint /Users/josh/repos/cta-temp-oct-27
│  > eslint . --max-warnings 0 --fix
│  
│  
│  /Users/josh/repos/cta-temp-oct-27/package.json
│    1:186  error  Invalid author: Type for field "author" should be a `string` or an `object` with at least a `name` property  package-json/valid-author
│  
│  ✖ 1 problem (1 error, 0 warnings)
│  
│  \u2009ELIFECYCLE\u2009 Command failed with exit code 1.
│      at getFinalError (file:///Users/josh/repos/create-typescript-app/node_modules/.pnpm/execa@9.6.0/node_modules/execa/lib/return/final-error.js:6:9)
│      at makeError (file:///Users/josh/repos/create-typescript-app/node_modules/.pnpm/execa@9.6.0/node_modules/execa/lib/return/result.js:108:16)
│      at getAsyncResult (file:///Users/josh/repos/create-typescript-app/node_modules/.pnpm/execa@9.6.0/node_modules/execa/lib/methods/main-async.js:168:4)
│      at handlePromise (file:///Users/josh/repos/create-typescript-app/node_modules/.pnpm/execa@9.6.0/node_modules/execa/lib/methods/main-async.js:151:17)
│      at async runCommand (file:///Users/josh/repos/create-typescript-app/node_modules/.pnpm/bingo@0.5.16/node_modules/bingo/lib/runners/applyScriptsToSystem.js:10:24)
│      at async file:///Users/josh/repos/create-typescript-app/node_modules/.pnpm/bingo@0.5.16/node_modules/bingo/lib/runners/applyScriptsToSystem.js:22:17
│      at async Promise.all (index 1)
│      at async applyScriptsToSystem (file:///Users/josh/repos/create-typescript-app/node_modules/.pnpm/bingo@0.5.16/node_modules/bingo/lib/runners/applyScriptsToSystem.js:20:9)
│      at async Promise.all (index 0)
│      at async runCreation (file:///Users/josh/repos/create-typescript-app/node_modules/.pnpm/bingo@0.5.16/node_modules/bingo/lib/runners/runCreation.js:8:5)
│
└  Thanks for using create-typescript-app! 💝
```

### Additional Info

The `author.name` field is not populated. It normally comes from:

https://github.com/JoshuaKGoldberg/create-typescript-app/blob/855c1dafe6d9030a2f3a29e46e11b1da807b6b0b/src/options/readNpmDefaults.ts#L8-L13

That which calls to the `npm-user` package:

https://github.com/JoshuaKGoldberg/create-typescript-app/blob/afc398b0f27c6b6e8699e7e0d471c64438bbac04/src/base.ts#L249-L256

...which is seeing 403s from the npm website: https://github.com/sindresorhus/npm-user/issues/24

❤️‍🔥

--- a/src/base.ts
+++ b/src/base.ts
@@ -177,7 +177,12 @@ export const base = createBase({
 
 		const getAuthor = lazyValue(
 			async () =>
-				await readAuthor(getPackageAuthor, getNpmDefaults, options.owner),
+				await readAuthor(
+					getPackageAuthor,
+					getNpmDefaults,
+					getGitUser,
+					options.owner,
+				),
 		);
 
 		const getBin = lazyValue(async () => await readBin(getPackageData));
@@ -235,6 +240,11 @@ export const base = createBase({
 		const getFunding = lazyValue(async () => await readFunding(take));
 
 		const getGitDefaults = lazyValue(async () => await readGitDefaults(take));
+
+		const getGitUser = lazyValue(
+			async () =>
+				await take(inputFromScript, { command: "git config user.name" }),
+		);
 
 		const getGuide = lazyValue(async () => await readGuide(take));
 

--- a/src/options/readAuthor.test.ts
+++ b/src/options/readAuthor.test.ts
@@ -1,3 +1,4 @@
+import { Result } from "execa";
 import { describe, expect, it, vi } from "vitest";
 
 import { readAuthor } from "./readAuthor.js";
@@ -10,6 +11,7 @@ describe(readAuthor, () => {
 		const actual = await readAuthor(
 			() => Promise.resolve({ name }),
 			getNpmDefaults,
+			() => Promise.resolve(undefined),
 			undefined,
 		);
 
@@ -23,6 +25,7 @@ describe(readAuthor, () => {
 		const actual = await readAuthor(
 			() => Promise.resolve({}),
 			() => Promise.resolve({ name }),
+			() => Promise.resolve(undefined),
 			undefined,
 		);
 
@@ -35,15 +38,30 @@ describe(readAuthor, () => {
 		const actual = await readAuthor(
 			() => Promise.resolve({}),
 			() => Promise.resolve(undefined),
+			() => Promise.resolve(undefined),
 			owner,
 		);
 
 		expect(actual).toBe(owner);
 	});
 
+	it("returns gitUser when only it exists", async () => {
+		const gitUser = "test-owner";
+
+		const actual = await readAuthor(
+			() => Promise.resolve({}),
+			() => Promise.resolve(undefined),
+			() => Promise.resolve({ stdout: gitUser } as Result),
+			undefined,
+		);
+
+		expect(actual).toBe(gitUser);
+	});
+
 	it("returns undefined when no sources provide a value", async () => {
 		const actual = await readAuthor(
 			() => Promise.resolve({}),
+			() => Promise.resolve(undefined),
 			() => Promise.resolve(undefined),
 			undefined,
 		);

--- a/src/options/readAuthor.ts
+++ b/src/options/readAuthor.ts
@@ -1,11 +1,17 @@
+import { ExecaError, Result } from "execa";
+
 import { PackageAuthor } from "./readPackageAuthor.js";
 
 export async function readAuthor(
 	getPackageAuthor: () => Promise<PackageAuthor>,
 	getNpmDefaults: () => Promise<undefined | { name?: string }>,
+	getGitUser: () => Promise<ExecaError | Result | undefined>,
 	owner: string | undefined,
 ) {
 	return (
-		(await getPackageAuthor()).name ?? (await getNpmDefaults())?.name ?? owner
+		(await getPackageAuthor()).name ??
+		(await getNpmDefaults())?.name ??
+		(await getGitUser())?.stdout?.toString() ??
+		owner
 	);
 }

